### PR TITLE
Webjobs

### LINKED ISSUE #2: Add support for WebJob endpoints
Please consider adding support for the Kudu WebJob API endpoints:

https://github.com/projectkudu/kudu/wiki/WebJobs-API

### PR DESCRIPTION
**Pull request by benquarmby**
This pull request contains methods for all the webjob endpoints documented here:
https://github.com/projectkudu/kudu/wiki/WebJobs-API

There are some errors in the documentation, but I was able to extrapolate around it. I also happen to know there are undocumented APIs like listing "functions" for jobs using the Azure SDK for .NET. But this seemed like a good start and worth bringing in as is.

This resolves #2.

### CODE REVIEW

File: lib/webjobs.js [hunk 1]
```diff
@@ -1,48 +1,70 @@
 "use strict";
 
+function resolveError(response, message) {
```
Comment (benquarmby) [inline lib/webjobs.js]: It's possible this kind of generalized error handling could help elsewhere.

File: test/webjobs.js [hunk 2]
```diff
@@ -1,54 +1,174 @@
...
+var JSZip = require("jszip");
 var api = require("../")({website: process.env.WEBSITE, username: process.env.USERNAME, password: process.env.PASSWORD});
 
+function createJobZip(localPath, cb) {
+    var generateOptions = {
+        type: "nodebuffer",
+        streamFiles: true
+    };
+
+    var zip = new JSZip();
```
Comment (benquarmby) [inline test/webjobs.js]: I think we should use JSZip for all zip file creation. Means there is no need to rely on zip at the command line (which I didn't have).

File: test/webjobs.js [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+    zip.generateNodeStream(generateOptions)
+        .pipe(fs.createWriteStream(localPath))
+        .on("error", cb)
+        .on("finish", cb);
+}
+
+function createPollingCallback(cb) {
+    var operation = retry.operation({
+        retries: 5
+    });
```
Comment (benquarmby) [inline test/webjobs.js]: Might seem like overkill, but I really wanted to make sure these tests run first-time-every-time. Hopefully this eliminates any flakiness.

**Comment by benquarmby** [comment]
I will merge this in the next couple of days and follow up with a minor release soon after.

**Comment by itsananderson** [comment]
This is great! :+1:

**Pull request merged**